### PR TITLE
Add mkdocs site

### PR DESCRIPTION
**Pull request by kgang**
Edited by Michael:
 - Add docs Dockerfile/service
 - Add mkdocs config
 - Add some documentation for mkdocs
 - Update container port definitions in AwsTool to be dynamic

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -70,7 +86,21 @@ services:
...
+  docs:
+    build:
+      context: ../
+      dockerfile: ./docker/Dockerfile.docs
+    container_name: docs
+    env_file:
+      - path: "../.env"
+    image: docs
+    network_mode: service:ts-oauth-docs
+    ports:
```
Comment (MichaelDeng03) [inline docker/docker-compose.yml:99]: you'll have to remove ports here

File: .github/workflows/docs_staging.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                  aws-region: us-east-1
+
+            - name: Deploy to ECS staging using AwsTool._deploy_service
+              env:
+                image_uri: '008971649127.dkr.ecr.us-east-1.amazonaws.com/webapp-homepage:latest'
+              run: |
+                  curl -sSL https://install.python-poetry.org | python3 -
+                  poetry --version
+                  poetry install
+                  poetry run python -m concrete deploy --image_uri 008971649127.dkr.ecr.us-east-1.amazonaws.com/docs:latest --container_name webapp-homepage --container_port 80 --service_name=docs
```
[the remark this answers is not in the document]
Comment (kgang) [reply .github/workflows/docs_staging.yml:84]: omg noo

[the remark this answers is not in the document]
Comment (kgang) [reply .github/workflows/docs_staging.yml]: ty [](edited-with-Axolo)

[the remark this answers is not in the document]
Comment (MichaelDeng03) [reply .github/workflows/docs_staging.yml]: lol [](edited-with-Axolo)

**Review by kgang — COMMENTED** [review]
Thanks for removing the generated files LGTM

**Pull request merged**